lib/src/create_report.dart

```diff
@@ -745,14 +749,88 @@ Future<ReportSection> _trustworthyDependency(PackageContext context) async {
 
   final dependencySection = await dependencies();
   final sdkSection = await sdkSupport();
+  final subsections = [dependencySection, sdkSection];
+  return _makeSection(
+    id: ReportSectionId.dependency,
+    title: 'Support up-to-date dependencies',
+    maxPoints: 20,
+    subsections: subsections,
+    basePath: packageDir,
+  );
+}
 
-  return ReportSection(
-      id: ReportSectionId.dependency,
-      title: 'Support up-to-date dependencies',
-      grantedPoints: dependencySection.grantedPoints + sdkSection.grantedPoints,
-      maxPoints: 20,
-      summary:
-          _makeSummary([dependencySection, sdkSection], basePath: packageDir));
+_Issue _explanationToIssue(Explanation explanation) =>
+    _Issue(explanation.finding, suggestion: explanation.explanation);
+
+Future<ReportSection> _nullSafety(String packageDir, Pubspec pubspec) async {
+  // TODO(sigurdm): Currently we don't give any points for null-safety.
+  // Raise this after null-safety beta.
+  const maxPoints = 0;
+
+  _Subsection subsection;
+  if (File(p.join(packageDir, '.dart_tool', 'package_config.json'))
+      .existsSync()) {
+    final tagger = Tagger(packageDir);
+
+    final nullSafetyTags = <String>[];
+    final explanations = <Explanation>[];
+    tagger.nullSafetyTags(nullSafetyTags, explanations);
+    if (pubspec.sdkConstraintStatus.hasOptedIntoNullSafety) {
+      if (nullSafetyTags.contains('is:null-safe')) {
+        subsection = _Subsection(
+            'Package and dependencies are fully migrated to null-safety',
+            explanations.map(_explanationToIssue).toList(),
+            maxPoints,
+            maxPoints,
+            ReportStatus.passed);
+      } else {
+        subsection = _Subsection(
+            'Package declares support for null-safety, but there are issues',
+            [
+              ...explanations.map(_explanationToIssue).toList(),
+              _Issue('For more information',
+                  suggestion: 'Try running `pub outdated --mode=null-safety`.\n'
+                      'Be sure to read the [migration guide](https://dart.dev/null-safety/migration-guide).')
```

Comment:
This will no longer give the full picture..

Let's add a TODO to explain it here...

Reply:
Done